plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Repository.java

```diff
@@ -190,6 +190,13 @@ class S3Repository extends MeteredBlobStoreRepository {
         new ByteSizeValue(5, ByteSizeUnit.TB)
     );
 
+    /**
+     * Maximum number of deletes in a DeleteObjectsRequest.
+     *
+     * @see <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/multiobjectdeleteapi.html">S3 Documentation</a>.
+     */
+    static final Setting<Integer> BULK_DELETE_SIZE = Setting.intSetting("bulk_delete_size", 1000, 1, 1000);
```

Comment:
Lets have minimum as 0 to allow deletes to not happen if required by the user?

Reply:
0 is a breach of contract for the API, hence not going ahead with that.